### PR TITLE
Data and networking

### PR DESCRIPTION
**Pull request by BenGoldstein88**

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -1,13 +1,16 @@
+@import url("https://fonts.googleapis.com/css?family=Open+Sans");
+@import url("https://fonts.googleapis.com/css?family=Spartan");
+@import url("https://fonts.googleapis.com/css?family=Caladea");
 body {
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
-    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
```
Comment (samgold89) [inline src/index.css:6]: FYI Nunito is the go to font, for whenever that becomes relevant.

File: src/pages/UsersAndTeams.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ export const UsersAndTeams = props => {
   };
 
   const updateFromCsv = users => {
-    const uniqueTeams = new Set(users.map(user => user.name));
+    const uniqueTeams = [...new Set(users.map(user => user.name))];
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:92]: <3

File: src/pages/UsersAndTeams.js [hunk 3]
```diff
@@ -179,13 +185,69 @@ export const UsersAndTeams = props => {
       });
   };
 
+  const editUser = userInfo => {
+    const { id, firstName, lastName, email, teamId } = userInfo;
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:189]: `teamId` isn't needed to update a User, but maybe this is just a react thing I don't get :)

File: src/pages/UsersAndTeams.js [hunk 4]
```diff
@@ -179,13 +185,69 @@ export const UsersAndTeams = props => {
...
+    const url = buildUrl({
+      eventSlug,
+      path: `users/${id}`
+    });
+    fetch(proxyUrl + url, {
+      method: "PUT",
+      headers: {
+        ...devHeaders,
+        "Content-Type": "application/json",
+        Accept: "application/json"
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:199]: Shouldn't all headers pull in from some constant somewhere ... ? They'll all be the same.

File: src/pages/UsersAndTeams.js [hunk 5]
```diff
@@ -179,13 +185,69 @@ export const UsersAndTeams = props => {
...
+    fetch(proxyUrl + url, {
+      method: "POST",
+      headers: {
+        ...devHeaders,
+        "Content-Type": "application/json",
+        Accept: "application/json"
+      },
+      body: JSON.stringify({
+        reset_auth_token: true,
+        userId
```
Comment (samgold89) [inline src/pages/UsersAndTeams.js:231]: oh this is cool. So how does it know which key to use in this dict?

File: src/pages/UsersAndTeams.js (hunk 3 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:189]: Allows for component reuse between EDIT and CREATE :)

File: src/pages/UsersAndTeams.js (hunk 4 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:199]: I didn't need content-type/accept for GETs so they're added to PUT/POST, but you're right i can abstract 'em out

File: src/pages/UsersAndTeams.js (hunk 5 above)
Comment (BenGoldstein88) [reply src/pages/UsersAndTeams.js:231]: If you have a variable in scope called (e.g.) userId, then if you create this object---- { userId } ---- it reads it as ----  { userId: userId } ----.  In this case, userId is destructured and passed as param. 

That is to say, the whatever calls joinUserToTeam(obj), the obj will look like { userId: "something", teamId: "otherthing" }.

File: src/index.css (hunk 1 above)
Comment (BenGoldstein88) [reply src/index.css:6]: 👍

**Pull request merged**